tests/nvidiagpu/deploy-gpu-test.go

```diff
@@ -371,54 +231,41 @@ var _ = Describe("GPU", Ordered, Label(tsparams.LabelSuite), func() {
 
 		AfterAll(func() {
 
-			if nfdCleanupAfterInstall && cleanupAfterTest {
+			if Nfd.CleanupAfterInstall && cleanupAfterTest {
 				// Here need to check if NFD CR is deployed, otherwise Deleting a non-existing CR will throw an error
 				// skipping error check for now cause any failure before entire NFD stack
 				By("Delete NFD CR instance in NFD namespace")
-				_ = deploy.NFDCRDeleteAndWait(inittools.APIClient, nfdCRName, nfdOperatorNamespace, 30*time.Second, 5*time.Minute)
+				_ = nfd.NFDCRDeleteAndWait(inittools.APIClient, nfd.CRName, nfd.OperatorNamespace, nvidiagpu.DeletionPollInterval, nvidiagpu.DeletionTimeoutDuration)
 
 				By("Delete NFD CSV")
-				_ = deploy.DeleteNFDCSV(inittools.APIClient)
+				_ = nfd.DeleteNFDCSV(inittools.APIClient)
 
 				By("Delete NFD Subscription in NFD namespace")
-				_ = deploy.DeleteNFDSubscription(inittools.APIClient)
+				_ = nfd.DeleteNFDSubscription(inittools.APIClient)
 
 				By("Delete NFD OperatorGroup in NFD namespace")
-				_ = deploy.DeleteNFDOperatorGroup(inittools.APIClient)
+				_ = nfd.DeleteNFDOperatorGroup(inittools.APIClient)
 
 				By("Delete NFD Namespace in NFD namespace")
-				_ = deploy.DeleteNFDNamespace(inittools.APIClient)
+				_ = nfd.DeleteNFDNamespace(inittools.APIClient)
 			}
```

Comment:
All this functionality should be in the NFD package, GPU/NNO tests shouldn't care what a cleanup means for NFD, and will just "ask" it to clean up itself